libtransmission/rpc-server.cc

```diff
@@ -999,39 +1167,57 @@ tr_rpc_server::tr_rpc_server(tr_session* session_in, tr_variant* settings)
         tr_rpcSetAntiBruteForceThreshold(this, i);
     }
 
+    key = TR_KEY_rpc_socket_mode;
+
+    if (!tr_variantDictFindInt(settings, key, &i))
+    {
+        missing_settings_key(key);
+    }
+    else
+    {
+        tr_rpcSetRPCSocketMode(this, i);
+    }
+
     key = TR_KEY_rpc_bind_address;
 
     if (!tr_variantDictFindStrView(settings, key, &sv))
     {
         missing_settings_key(key);
-        address = tr_inaddr_any;
+        address = tr_rpc_inaddr_any;
     }
     else
     {
-        if (!tr_address_from_string(&address, std::string{ sv }.c_str()))
+        if (!tr_rpc_address_from_string(&address, sv))
         {
-            tr_logAddNamedError(MyName, _("%" TR_PRIsv " is not a valid address"), TR_PRIsv_ARG(sv));
-            address = tr_inaddr_any;
+            tr_logAddNamedError(MyName, _("%" TR_PRIsv " is not a valid address, falling back to 0.0.0.0."), TR_PRIsv_ARG(sv));
+            address = tr_rpc_inaddr_any;
         }
-        else if (address.type != TR_AF_INET && address.type != TR_AF_INET6)
+        else if (address.type != TR_RPC_AF_INET && address.type != TR_RPC_AF_INET6 && address.type != TR_RPC_AF_UNIX)
```

Comment:
Looks like it's worth introducing a function for this check, which could then also be used in `tr_rpc_address_is_valid()` to avoid code duplication. Alternatively, I don't see any way for `tr_rpc_address_from_string()` to return `true` and at the same time set `type` to something other than one of the valid values, so I'm not sure we even need this check here.

Reply:
This was something else that was copied over from the previous implementation, though you're right there should be no way for this to be reached.